CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,149 @@
+       1. Failure to adapt can lead to an unreasonable cost to the community to maintain membership.
+       2. This may create situations which may seem contradictory. I understand however that this is in recognition of Code of Conduct item 4.
+
+</details>
+
+4. <details><summary>We understand the expectations of us to act according to values and not technicalities.</summary><br />
+
+    1. The intolerable will never be tolerable. We will play by the rules as intended. Where it is not seen as discriminatory, people who try to excuse poor behaviour as not being written clearly within the Code of Conduct will be treated as if it was
+       1. As an example: Discussions such as questioning the legitimacy of genocidal events in history are inexcusable. It being framed as a question or due to ones own ignorance will be investigated under the *reasonable person principle* and punished appropriately.
+          1. We do not expect situations like this to arise; our conversation should mostly be productive; however, we are an enjoyable group of people who will no doubt discuss different aspects of life.
+
+</details>
+
+5. <details><summary>Contribution implicitly registers acknowledgement of this Code of Conduct.</summary><br />
+
+    1. Ignorance of the rules is not a valid argument for breaking them.
+
+</details>
+
+6. <details><summary>I/We will work towards keeping this environment productive by ensuring behaviour is correct to this Code of Conduct.</summary><br />
+
+    1. If you see someone being harassed, speak up. We will support you, and whole heartedly thank you.
+    2. Silence can sometimes be worse than agreeing with a harassor/perpetrator.
+    3. If you are concerned due to some connection to the aggressor and do not wish to alienate yourselves from them, please contact us and we will handle the situation with complete confidentiality.
+       1. Please see the code for enforcement below.
+
+</details>
+
+<p align="right">(<a href="#top">back to top</a>)</p>
+
+## Enforcement
+
+When enforcing the Code of Conduct, especially in cases involving non-native English speakers, we encourage patience, empathy, understanding, and willingness to educate and inform. In context where the breach is deemed *intentional*, we will pursue enforcement of the Code of Conduct as described.
+
+> *"People from different backgrounds are likely to have different interpretations of what words or behaviors are considered acceptable. It's somewhat ironic that in an attempt to draw a more diverse developer base, code of conduct moderation, if improperly conducted, could potentially alienate users from other countries who don't have the same social conditioning as those in North America [or other Western cultures]."*[^2]
+
+1. The Code of Conduct can be enforced through a range of passive and active means. We recommend minimal use for first offenders but allow rapid escalation if required (depending on the severity of the breach).
```

Comment:
What is "minimal use"?

Reply:
As according to the reasonable person principle; we have not yet discussed hierarchy of rule enforcement so I cannot determine this without input from everyone.

Reply:
I'm ok with it, gives room to grow.